### PR TITLE
Request handler refactor branch

### PR DESCRIPTION
**Pull request by timqy**
refactored request handler

### CODE REVIEW

**Comment by 23Skidoo** [comment]
Needs to be updated to merge cleanly.

**Comment by 23Skidoo** [comment]
Looks OK on a first glance.

**Comment by timqy** [comment]
if we were to change it back to the previous 'performAuthorization'
i would suggest renaming it to perfomAuthentication instead, because it's a function that determines who you are(authentication), not what you are allowed to do (Authorization).

**Comment by currybullen** [comment]
The name has now been changed to performAuthentication and it should also be prepared for auto-merging.

**Comment by 23Skidoo** [comment]
Still doesn't merge cleanly.

**Comment by 23Skidoo** [comment]
We should also probably parse the query only once at top level and pass in `HashMap<String, String>` instead of a `String`.

**Comment by 23Skidoo** [comment]
> We should also probably parse the query only once at top level and pass in `HashMap<String, String>` instead of a `String`.

Implemented in #288.

File: src/authentication/Authenticate.java [hunk 1]
```diff
@@ -125,4 +127,48 @@ static public String getUsernameByID(String userID){
 		return latestRequests;
 	}
 
+	/**
+	 * Performs authorization, returns null if the user could not be authorized,
```
Comment (23Skidoo) [inline src/authentication/Authenticate.java]: The comment should also use "authentication" instead of "authorization".

**Pull request merged**